azext_baremetalinfrastructure/custom.py

```diff
@@ -5,24 +5,22 @@
 
 # pylint: disable=no-self-use,too-many-lines
 
-
 def show_baremetalinstance(client, resource_group_name, instance_name):
     return client.get(resource_group_name, instance_name)
 
-
 def list_baremetalinstance(client, resource_group_name=None):
     if resource_group_name is None:
         return client.list_by_subscription()
     return client.list(resource_group_name)
 
-def restart_baremetalinstance(client, resource_group_name, instance_name):
-    # The restart baremetalinstance REST API is a POST with no body.
+def restart_baremetalinstance(client, resource_group_name, instance_name, force=False):
+    # The restart baremetalinstance REST API is a POST with a body.
     # The HaaS RP API requires the Content-Type to be set.
     # Swagger does not add the Content-Type in the header if there is no body.
     # We need to add a custom header to force the API call to add the Content-Type.
     custom_header = {}
-    custom_header['Content-Type'] = 'application/json; charset=utf-8'
```

Comment:
are this not necessary anymore?

Reply:
The custom header is used in the `begin_restart` function, who is calling `_restart_initial`. Inside restart initial we have the following code:

![image](https://github.com/Azure/azure-baremetalinfrastructure-cli-extension/assets/26116542/38bb96bd-bc5e-40f0-85a2-11e681459713)

meaning that for empty object we by default are going to be granted the original "application/json".